### ISSUE TITLE
4 Channel RGBW support

### ISSUE THREAD

**Issue by licensed2hench** [smeighan/xLights#2302]
Problem:
I am using 4 Channel RGBW pixels with partial success. 
Of all the various String Type options only "4 Channel RGB" works. 
However, the Red and Green Channels are swapped.

Requested Solution:
Have the 2 "4 Channel RGBW" options expanded to offer more color order options.
"4 Channel GRBW" - etc.

Alternatives:
I have tried all the various String Type options but no other variation worked.
The Controller: Set Color Order option didn't work with my original controller, so I got a PiCAP. Unfortunately the PiCAP only supports color reordering on a 3 channel scheme. (which caused some truly strange results) I can output to the PiCAP I just cant fix the color swap.

Additional:
I am using a NeoPixel Jewel RGBW LED PRODUCT ID: 2858
It may be using the ws2812 protocol? I am unsure. ws2811 works fine.

**Comment by Kamal-Sonani** [comment]
Drop your element in canvas

Unfold appearance tab ( left bottom)

Select color order 
Feature was Already loaded:v:

**Comment by licensed2hench** [comment]
Thanks for the video link - it is the kind of introductory tutorial I need :smile: 

The issue I have is on the String Properties Tab of my model
Currently I am using:
_String Type:_ **4 Channel RGBW**
_RGBW Color Handling:_ **R=G=B->W**

This works with my 4 color pixels (GRBW)
However the Red and Green channels are reverse (so it lights Red instead of Green and Green instead of Red)

The only other String Type option that works at all is _4 Channel WRGB_ but then the even more of the color channels are swapped. :frowning_face: 

What I need is to have some additional **4 Channel x** String Type options added to cover all the various color order combinations. (Much like the various _RGBW Nodes_ options)

Current _String Type_ options available:
4 Channel RGBW
4 Channel WRGB

Additional _String Type_ options requested:
4 Channel RBGW
4 Channel GBRW
**4 Channel GRBW** - (This is the one I specifically need to fix my issue)
4 Channel BRGW
4 Channel BGRW
4 Channel WRBG
4 Channel WGBR
4 Channel WGRB
4 Channel WBRG
4 Channel WBGR

Thanks

**Comment by keithsw1111** [comment]
Shouldnt you be using "GRBW Nodes" which already exists?

**Comment by licensed2hench** [comment]
_GRBW Nodes_ doesn't work. It shifts the colors around differently for each pixel. 
I have tried every combination of String Type and Color Handling options. All with the jumbled color results.
I guess the real question is why does the _4 Channel RBGW_ option work at all? 😉 
Whatever is different about it, I need more of that.

**Comment by dkulp** [comment]
The GRBW Nodes seems to be working correctly to me.   If I create a shadow model of single color nodes, you can see the raw channels.    If I drop Green on the main GRBW model, every 4th channel lights up starting with the first.   This looks completely correct to me.  

<img width="731" alt="Screen Shot 2020-10-23 at 3 36 54 PM" src="https://user-images.githubusercontent.com/91569/97046690-be9fc300-1545-11eb-9d9f-041f3dbeea74.png">

**Comment by licensed2hench** [comment]
Am I using GRBW Nodes wrong?
Is each node a separate led in the pixel?
So in my case a string of 7, GRBW pixels would be 28 nodes?

I had been assuming that a node=pixel so 7 nodes per string.
I will do more testing.

**Comment by dkulp** [comment]
A "Node" in this case would be 4 channels.    So a string of 7 "pixels" would be 28 channels.   

In my screen shot, that would be "Single-Line".    The "Single-Line 3" model is a shadow model that starts at the same start channel, but is a bunch of "Single Channel" nodes just so I can see exactly what the data is at each channel.

**Comment by licensed2hench** [comment]
I made a "Shadow String" of 28 single color white nodes starting at the same channel.
I could not get it to display the signal from the other Model so I could not "see" what it was outputting. 😦 

I tried sending signal to each node separately.
On the shadow node it worked as expected. By sending On: White, to each node separately, I could light each pixel one color at a time, in the expected order.
(G.R.B.W.  for all 7 pixels)

On the regular model: Single strand, (7) GRBW Nodes, R=G=B->W (using "Advanced" yielded the same results)
Things got weird.
Node 1: Green = Pixel 1: Green
Node 1: Red = Pixel 1: Red
Node 1: Blue = Pixel 1: Blue
Node 1: White = Pixel 1: White (so far so good)

Node 2: Green = Pixel 1: White (wait what?)
Node 2: Red = Pixel 2: Green
Node 2: Blue = Pixel 2: Red
Node 2: White = Pixel 2: Blue

Node 3: Green = Pixel 2: Blue (ok I think I see what is happening)
Node 3: Red = Pixel 2: White
Node 3: Blue = Pixel 3: Green
Node 3: White = Pixel 3: Red

Node 4: Green = Pixel 3: Red (yep a definite pattern)
Node 4: Red = Pixel 3: Blue
Node 4: Blue = Pixel 3: White
Node 4: White = Pixel 4: Green

Node 5: Green = Pixel 4: Green
Node 5: Red = Pixel 4: Red
Node 5: Blue = Pixel 4: Blue
Node 5: White = Pixel 4: White

Node 6: Green = Pixel 4: White
Node 6: Red = Pixel 5: Green
Node 6: Blue = Pixel 5: Red
Node 6: White = Pixel 5: Blue

Node 7: Green = Pixel 5: Blue
Node 7: Red = Pixel 5: White
Node 7: Blue = Pixel 6: Green
Node 7: White = Pixel 6: Red

Ok so while each node is outputting 4 channels, it is like it is not calculating the start channel for the next node properly. (like it is adding add +3 to the previous nodes end address instead of +4?)
I have no idea what might be causing this?
Using the Shadow string I can successfully manually address each channel as expected so I don't _think_ it is any weirdness in the controller.

I am using ws2811 protocol.  I will play with others to see if that is an issue?

If you can give more info on how to set up a shadow model to be able to see what is actually being output to each channel that will help.

Thanks

**Comment by licensed2hench** [comment]
Looking closely I can see the data on the shadow channel it is just display dark grey on darker grey (sorry I could not find the color setting to change it) it looks like it is outputting every 3 channels for me? What am I doing different?
![image](https://user-images.githubusercontent.com/17708431/97067799-34bc1e00-1576-11eb-81ca-81c197e28c3c.png)

**Comment by licensed2hench** [comment]
A screen shot of the Model Settings:

![image](https://user-images.githubusercontent.com/17708431/97067825-9e3c2c80-1576-11eb-9a69-15c58827f6d6.png)

**Comment by licensed2hench** [comment]
Using the String type: **4 Channel RGBW**
It does output every 4 channels, just the Red and Green channel are swapped. 🤷‍♂️ 

![image](https://user-images.githubusercontent.com/17708431/97068022-b14ffc00-1578-11eb-8686-d1451fca8f8d.png)

**Comment by keithsw1111** [comment]
Attached is this implemented both with GRBW nodes and also with superstrings ... both work fine.

[test.zip](https://github.com/smeighan/xLights/files/5432535/test.zip)

**Comment by licensed2hench** [comment]
Thank you ! this was incredibly helpful.

First off - superstring works! (I never thought to try that as I had no idea what it was)
Also GRBW Nodes - works as expected on **Strings**!
So that fixed my personal issue, and gave me a viable work around, but I did find a bug!

GRBW Nodes does not properly on my **Custom Model**, or on any new **Custom Model** that I create.
I have included a side by side example of what I am seeing.
[grbw-test.zip](https://github.com/smeighan/xLights/files/5432672/grbw-test.zip)

Thanks on helping me to figure this out.
